### PR TITLE
persistent-qq: Fix Haddock markup + fix warnings + refactor

### PR DESCRIPTION
**Pull request by sol**
This are all the commits from #1111 that don't change any behavior; as a separate PR.  I assume these require less discussion + they are useful on their own.  So let's get these merged first!

### CODE REVIEW

**Comment by sol** [comment]
@parsonsmatt @snoyberg ping.

File: persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs [hunk 1]
```diff
@@ -69,6 +70,21 @@ parseStr a ('^':'{':xs) = Literal (reverse a) : parseHaskell TableName  [] xs
 parseStr a ('@':'{':xs) = Literal (reverse a) : parseHaskell ColumnName [] xs
 parseStr a (x:xs)       = parseStr (x:a) xs
 
+interpolateValues :: PersistField a => NonEmpty a -> (Text, [[PersistValue]]) -> (Text, [[PersistValue]])
+interpolateValues xs =
+    first (mkPlaceholders values <>) .
+    second (NonEmpty.toList values :)
+    where
```
Comment (parsonsmatt) [inline persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs:77]: Minor style comment 
```suggestion
  where
```

File: persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs [hunk 2]
```diff
@@ -69,6 +70,21 @@ parseStr a ('^':'{':xs) = Literal (reverse a) : parseHaskell TableName  [] xs
...
+    values = NonEmpty.map toPersistValue xs
+
+mkPlaceholders :: NonEmpty a -> Text
+mkPlaceholders values = "(" <> n `timesCommaSeparated` "?" <> ")"
+    where
```
Comment (parsonsmatt) [inline persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs:82]:
```suggestion
  where
```

File: persistent-qq/test/Spec.hs [hunk 3]
```diff
@@ -19,33 +19,27 @@ import PersistTestPetType
...
-sqlite_database = mkSqliteConnectionInfo sqlite_database_file
-
 runConn :: MonadUnliftIO m => SqlPersistT (LoggingT m) t -> m ()
 runConn f = do
-  let debugPrint = _debugOn
-  let printDebug = if debugPrint then print . fromLogStr else void . return
+  let printDebug = when _debugOn . B.putStrLn . fromLogStr
   flip runLoggingT (\_ _ _ s -> printDebug s) $ do
-    _ <- withSqlitePoolInfo sqlite_database 1 $ runSqlPool f
+    _ <- withSqliteConn ":memory:" $ runSqlConn f
```
Comment (parsonsmatt) [inline persistent-qq/test/Spec.hs:31]: What motivated this change?

Comment (sol) [reply persistent-qq/test/Spec.hs:31]: @parsonsmatt creating a file in the source tree interferes with file watching (specifically, `hspec/sensei`).  If you don't like the in-memory approach, I can create the sqlite-file somewhere under `/tmp` instead.

File: persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs (hunk 2 above)
Comment (sol) [reply persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs]: @parsonsmatt I followed the odd indentation for `where` to match the style used in the rest of the file.  I pushed 
e12934c to fix it everywhere within that source file.  Feel free to cherry-pick.

File: persistent-qq/test/Spec.hs (hunk 3 above)
Comment (parsonsmatt) [reply persistent-qq/test/Spec.hs:31]: It's totally fine, I'm just curious for the change. In memory seems like it'd be preferable anyway, and I'm not sure why it was setup like that.

File: persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs (hunk 2 above)
Comment (parsonsmatt) [reply persistent-qq/src/Database/Persist/Sql/Raw/QQ.hs]: Ah! Thank you. I appreciate it <3

**Comment by sol** [comment]
@parsonsmatt an other friendly ping.

**Pull request merged**